python/anvill/binja.py

```diff
@@ -30,80 +30,89 @@ def is_valid_addr(bv, addr):
         bn.LowLevelILOperation.LLIL_CONST_PTR,
     )
 
 
-def is_constant_pointer(bv, insn):
-    return insn.operation == bn.LowLevelILOperation.LLIL_CONST_PTR
+def is_constant_pointer(bv, inst):
+    return inst.operation == bn.LowLevelILOperation.LLIL_CONST_PTR
 
 
-def is_function_call(bv, insn):
-    return insn.operation in (
+def is_function_call(bv, inst):
+    return inst.operation in (
         bn.LowLevelILOperation.LLIL_CALL,
         bn.LowLevelILOperation.LLIL_TAILCALL,
         bn.LowLevelILOperation.LLIL_CALL_STACK_ADJUST,
     )
 
 
-def is_tailcall(bv, insn):
-    return insn.operation == bn.LowLevelILOperation.LLIL_TAILCALL
+def is_tailcall(bv, inst):
+    return inst.operation == bn.LowLevelILOperation.LLIL_TAILCALL
 
 
-def is_return(bv, insn):
-    return insn.operation == bn.LowLevelILOperation.LLIL_RET
+def is_return(bv, inst):
+    return inst.operation == bn.LowLevelILOperation.LLIL_RET
 
 
-def is_jump(bv, insn):
-    return insn.operation in (
+def is_jump(bv, inst):
+    return inst.operation in (
         bn.LowLevelILOperation.LLIL_JUMP,
         bn.LowLevelILOperation.LLIL_JUMP_TO,
     )
 
 
-def is_branch(bv, insn):
-    return insn.operation in (
+def is_branch(bv, inst):
+    return inst.operation in (
         bn.LowLevelILOperation.LLIL_JUMP,
         bn.LowLevelILOperation.LLIL_JUMP_TO,
         bn.LowLevelILOperation.LLIL_GOTO,
     )
 
 
-def is_load_insn(bv, insn):
-    return insn.operation == bn.LowLevelILOperation.LLIL_LOAD
+def is_load_inst(bv, inst):
+    return inst.operation == bn.LowLevelILOperation.LLIL_LOAD
 
 
-def is_store_insn(bv, insn):
-    return insn.operation == bn.LowLevelILOperation.LLIL_STORE
+def is_store_inst(bv, inst):
+    return inst.operation == bn.LowLevelILOperation.LLIL_STORE
 
 
-def is_memory_insn(bv, insn):
-    return is_load_insn(bv, insn) or is_store_insn(bv, insn)
+def is_memory_inst(bv, inst):
+    return is_load_inst(bv, inst) or is_store_inst(bv, inst)
 
 
-def is_unimplemented(bv, insn):
-    return insn.operation == bn.LowLevelILOperation.LLIL_UNIMPL
+def is_unimplemented(bv, inst):
+    return inst.operation == bn.LowLevelILOperation.LLIL_UNIMPL
 
 
-def is_unimplemented_mem(bv, insn):
-    return insn.operation == bn.LowLevelILOperation.LLIL_UNIMPL_MEM
+def is_unimplemented_mem(bv, inst):
+    return inst.operation == bn.LowLevelILOperation.LLIL_UNIMPL_MEM
 
 
-def is_undef(bv, insn):
-    return insn.operation == bn.LowLevelILOperation.LLIL_UNDEF
+def is_undef(bv, inst):
+    return inst.operation == bn.LowLevelILOperation.LLIL_UNDEF
 
 
 def is_code(bv, addr):
-    sec_list = bv.get_sections_at(addr)
-    for sec in sec_list:
+    for sec in bv.get_sections_at(addr):
         if sec.start <= addr < sec.end:
             return sec.semantics == bn.SectionSemantics.ReadOnlyCodeSectionSemantics
     return False
 
 
+def is_data(bv, addr):
+    for sec in bv.get_sections_at(addr):
+        if sec.start <= addr < sec.end:
+            return sec.semantics in [
```

Comment:
Recommend using `== ... or == ..`, or making a global variable that is a tuple that has the two semantic types in them, otherwise every time we do this check, a list of two elements is created only to be destroyed later.